mrmustard/math/tensorflow.py

```diff
@@ -81,7 +81,10 @@ def clip(self, array, a_min, a_max) -> tf.Tensor:
         return tf.clip_by_value(array, a_min, a_max)
 
     def concat(self, values: Sequence[tf.Tensor], axis: int) -> tf.Tensor:
-        return tf.concat(values, axis)
+        try:
+            return tf.concat(values, axis)
+        except tf.errors.InvalidArgumentError:
+            return tf.stack(values, axis)
```

Comment:
This is obviously a hack. Either we are safe to replace all calls to `tf.concat` here with `tf.stack`, or we should likely be better about determining which to use based on the passed `values` rather than waiting for the `InvalidArgumentError`.

Reply:
ok, basing it on 0-rank seems to have fixed it.